prover/proof_search.py

```diff
@@ -383,7 +383,8 @@ def __init__(
 
         if with_gpus:
             logger.info(f"Launching {num_cpus} GPU workers.")
-            ray.init(num_cpus=num_cpus, num_gpus=num_cpus)
```

Comment:
No, this is not a typo. Here we want to pair each CPU with one GPU. We shouldn't use `torch.cuda.device_count()`, since the user may not want to use all GPUs on the machine (especially on shared clusters.)

Reply:
Then I should match num_cpus and num_gpus when I run the code..
Thanks!

Reply:
Then I have questions...
What happen when num_CPU > num_GPU..? 
When I tried, there are not error, but GPU's working is small...

Reply:
You only need to provide `num_cpus`. There is no parameter named `num_gpus`?